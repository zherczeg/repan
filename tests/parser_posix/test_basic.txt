# Test posix basic regular expressions

# Only * and [ are metacharacters, the result must be escaped by a backslash

/a*a\*/OB
    print

/a[b-d]**e\[f-h]**i/OB
    print

/a+a\+b?b\?/OB
    print

/(aa|bb)\(aa\|bb\)/OB
    print

/a{1,2}b\{1,2\}/OB
    print

/a\{1,2}/OB
    print

/a\{1,2\/OB
    print
