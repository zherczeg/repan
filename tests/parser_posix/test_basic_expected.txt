# Test posix basic regular expressions

# Only * and [ are metacharacters, the result must be escaped by a backslash

/a*a\*/OB
    print
    /a*a\*/

/a[b-d]**e\[f-h]**i/OB
    print
    /a(?:[b-d]*)*e\[f-h(?:]*)*i/

/a+a\+b?b\?/OB
    print
    /a\+a+b\?b?/

/(aa|bb)\(aa\|bb\)/OB
    print
    /\(aa\|bb\)(aa|bb)/

/a{1,2}b\{1,2\}/OB
    print
    /a\{1,2}b{1,2}/

/a\{1,2}/OB
    print
    Pattern parse failed: A backslash (\) is expected
        /a\{1,2[ HERE ]>>}/

/a\{1,2\/OB
    print
    Pattern parse failed: A right brace (}) is expected
        /a\{1,2\[ HERE ]>>/
